From b166295ef4124ce515a918bf6be7f5f9d9a87942 Mon Sep 17 00:00:00 2001
From: Eric Bentley <eric.bentley@lairdtech.com>
Date: Wed, 3 Feb 2016 23:33:11 -0500
Subject: [PATCH] laird: ath6kl: use add_krk_cmd size from firmware .h files

The size of the add_krk_cmd struct in both the 6003 and 6004 firmware
files is 26 but the driver had it set to 16.  The 6003 firmware did not
have issue with this but the 6004 firmware will fail to allow CCKM fast
roaming with the smaller size.  This change set the size to 26 to match
the firmware size.

bug 8853
---
 drivers/net/wireless/ath/ath6kl/wmi.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath6kl/wmi.h b/drivers/net/wireless/ath/ath6kl/wmi.h
index 617ae18..3015378 100644
--- a/drivers/net/wireless/ath/ath6kl/wmi.h
+++ b/drivers/net/wireless/ath/ath6kl/wmi.h
@@ -803,10 +803,11 @@ struct wmi_delete_cipher_key_cmd {
 } __packed;
 
 #define WMI_KRK_LEN     16
+#define WMI_CCKM_IE_LEN 26
 
 /* WMI_ADD_KRK_CMDID */
 struct wmi_add_krk_cmd {
-	u8 krk[WMI_KRK_LEN];
+	u8 krk[WMI_CCKM_IE_LEN]; //firmware wmi.h has size set to WMI_CCKM_IE_LEN
 } __packed;
 
 /* WMI_SETPMKID_CMDID */
-- 
1.9.1

